mldsa/src/sign.c

```diff
@@ -735,7 +743,11 @@ int crypto_sign_signature_extmu(uint8_t sig[MLDSA_CRYPTO_BYTES], size_t *siglen,
 
   /* Randomized variant of ML-DSA. If you need the deterministic variant,
    * call crypto_sign_signature_internal directly with all-zero rnd. */
-  mld_randombytes(rnd, MLDSA_RNDBYTES);
+  if (mld_randombytes(rnd, MLDSA_RNDBYTES) != 0)
+  {
+    *siglen = 0;
+    return -1;
```

Comment:
I am investigating integrating mldsa-native into TF-PSA-Crypto (formerly known as Mbed TLS). In our library, we would need to report the specific error code from the `randombytes` call (e.g. to distinguish a possibly temporary lack of entropy from a system integration failure). We aren't actually going to use `randombytes` at all because [we'll use `keypair_internal` and `signature_internal` anyway](https://github.com/pq-code-package/mldsa-native/issues/603#issuecomment-3672009944). However, it's likely that other users with similar system designs would want to retrieve the specific error code from their `randombytes` function.